src/lighteval/tasks/templates/qa.py

```diff
@@ -63,12 +63,13 @@ def get_qa_prompt_function(language: Language, adapter: Callable[[dict], QAInput
         Callable: A function that generates QA prompts based on the given parameters.
     """
 
-    adapter_fn: Callable[[dict], QAInput] = (
-        create_adapter_from_dict(adapter) if isinstance(adapter, dict) else adapter  # type: ignore
-    )
+    adapter_fn = create_adapter_from_dict(adapter)
```

Comment:
why the change ? the `create_adapter_from_dict` function did not change its typing.
also, the `adapter` var is not typed as dict in the function def

Reply:
<img width="435" alt="image" src="https://github.com/user-attachments/assets/521965dd-951d-4749-b96e-10d340e2a9b5">
I have made change to typing of that fc in some of the previous PRs, so this is a debt from previous PR. See the infered typing (there is no need to specify the type anymore)